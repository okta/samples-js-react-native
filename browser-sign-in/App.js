/*
 * Copyright (c) 2019-Present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import React, {Fragment} from 'react';

import {
  SafeAreaView,
  ScrollView,
  Button,
  StyleSheet,
  Text,
  View,
  StatusBar,
} from 'react-native';
import {
  createConfig,
  signInWithBrowser,
  signOut,
  getAccessToken,
  isAuthenticated,
  getUser,
  getUserFromIdToken,
  refreshTokens,
  EventEmitter,
} from '@okta/okta-react-native';

import configFile from './samples.config';

export default class App extends React.Component {
  constructor() {
    super();
    this.state = {
      authenticated: false,
      context: null,
    };
    this.checkAuthentication = this.checkAuthentication.bind(this);
    this.getMyUserThroughAccessToken = this.getMyUserThroughAccessToken.bind(
      this,
    );
  }

  async componentDidMount() {
    let that = this;
    EventEmitter.addListener('signInSuccess', function (error) {
      if (error) {
        console.warn(error);
        that.setContext(error.error_message);
        return;
      }

      that.setState({authenticated: true});
      that.setContext('Logged in!');
    });
    
    EventEmitter.addListener('signOutSuccess', function (error) {
      if (error) {
        console.warn(error);
        that.setContext(error.error_message);
        return; 
      }

      that.setState({authenticated: false});
      that.setContext('Logged out!');
    });

    EventEmitter.addListener('onError', function (error) {
      console.warn(error);
      that.setContext(error.error_message);
    });

    EventEmitter.addListener('onCancelled', function (error) {
      console.warn(error);
    });

    await createConfig({
      clientId: configFile.oidc.clientId,
      redirectUri: configFile.oidc.redirectUri,
      endSessionRedirectUri: configFile.oidc.endSessionRedirectUri,
      discoveryUri: configFile.oidc.discoveryUri,
      scopes: configFile.oidc.scopes,
      requireHardwareBackedKeyStore:
        configFile.oidc.requireHardwareBackedKeyStore,
    });
    this.checkAuthentication();
  }

  componentWillUnmount() {
    EventEmitter.removeAllListeners('signInSuccess');
    EventEmitter.removeAllListeners('signOutSuccess');
    EventEmitter.removeAllListeners('onError');
    EventEmitter.removeAllListeners('onCancelled');
  }

  async componentDidUpdate() {
    this.checkAuthentication();
  }

  async checkAuthentication() {
    const result = await isAuthenticated();
    if (result.authenticated !== this.state.authenticated) {
      this.setState({authenticated: result.authenticated});
    }
  }

  async login() {
    signInWithBrowser();
  }

  async logout() {
    signOut();
  }

  async getUserIdToken() {
    let user = await getUserFromIdToken();
    this.setContext(`
      User Profile:
      ${JSON.stringify(user, null, 4)}
    `);
  }

  async getMyUser() {
    let user = await getUser();
    this.setContext(`
      User Profile:
      ${JSON.stringify(user, null, 4)}
    `);
  }

  async getMyUserThroughAccessToken() {
    const accessToken = await getAccessToken();

    let headers = {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    };

    if (accessToken) {
      try {
        const wellKnown = await fetch(
          `${configFile.oidc.discoveryUri}/.well-known/openid-configuration`,
          {
            method: 'GET',
            headers: headers,
          },
        );
        if (!wellKnown.ok) {
          throw Error(wellKnown.status);
        }
        let json = await wellKnown.json();
        headers.Authorization = `Bearer ${accessToken.access_token}`;
        const userInfo = await fetch(`${json.userinfo_endpoint}`, {
          method: 'GET',
          headers: headers,
        });
        if (!userInfo.ok) {
          throw Error(userInfo.status);
        }
        json = await userInfo.json();
        this.setContext(JSON.stringify(json));
      } catch (e) {
        const message =
          'Failed to fetch user. Make sure you have logged in and access token is valid. Status Code: ' +
          e;
        console.warn(message);
        this.setContext(message);
      }
    } else {
      const message = 'There is no access token available!';
      console.warn(message);
      this.setContext(message);
    }
  }

  async refreshMyTokens() {
    let newTokens = await refreshTokens().catch(e => {console.log(e)});
    if (newTokens) {
    const message = ("Successfully refreshed tokens: " + JSON.stringify(newTokens));
    console.log(message);
    this.setContext(message);
    } 
    else {
     const message = ("Failed to refresh tokens: Be sure Refresh Token grant type is enabled in your app in Okta, as well as the offline_access scope in samples.config.js")
     this.setContext(message)
    }
  }

  setContext = (message) => {
    this.setState({
      context: message,
    });
  };

  renderButtons() {
    if (this.state.authenticated) {
      return (
        <View style={styles.buttonContainer}>
          <View style={styles.button}>
            <Button
              testID="getUserFromIdToken"
              onPress={async () => {
                this.getUserIdToken();
              }}
              title="Get User From Id Token"
            />
          </View>
          <View style={styles.button}>
            <Button
              testID="getUserFromRequest"
              onPress={async () => {
                this.getMyUser();
              }}
              title="Get User From Request"
            />
          </View>
          <View style={styles.button}>
            <Button
              testID="getMyUserFromAccessToken"
              onPress={async () => {
                this.getMyUserThroughAccessToken();
              }}
              title="Get User From Access Token"
            />
          </View>
          <View style={styles.button}>
            <Button
              testID="refreshMyTokens"
              onPress={async () => {
                this.refreshMyTokens();
              }}
              title="Refresh Tokens"
            />
          </View>
          <View style={styles.button}>
            <Button
              testID="clearButton"
              onPress={async () => {
                this.setContext('');
              }}
              title="Clear Text"
            />
          </View>
        </View>
      );
    }
  }

  render() {
    return (
      <Fragment>
        <StatusBar barStyle="dark-content" />
        <SafeAreaView style={styles.container}>
          <Text testID="titleLabel" style={styles.title}>Okta + React Native</Text>
          <View style={styles.buttonContainer}>
            <View style={styles.button}>
              {this.state.authenticated ? (
                <Button
                  style={styles.button}
                  testID="logoutButton"
                  onPress={async () => {
                    this.logout();
                  }}
                  title="Logout"
                />
              ) : (
                <Button
                  style={styles.button}
                  testID="loginButton"
                  onPress={async () => {
                    this.login();
                  }}
                  title="Login"
                />
              )}
            </View>
          </View>
          {this.renderButtons()}
          <ScrollView
            contentInsetAdjustmentBehavior="automatic"
            style={styles.context}>
            <Text testID="descriptionBox">{this.state.context}</Text>
          </ScrollView>
        </SafeAreaView>
      </Fragment>
    );
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginTop: 20,
  },
  button: {
    borderRadius: 40,
    height: 40,
    marginTop: 10,
    marginBottom: 10,
    marginHorizontal: 10,
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#FFFFFF',
    alignItems: 'center',
  },
  context: {
    marginTop: 20,
  },
  title: {
    fontSize: 30,
    fontWeight: 'bold',
    color: '#0066cc',
    paddingTop: 40,
    textAlign: 'center',
  },
});
