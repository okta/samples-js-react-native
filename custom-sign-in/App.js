/*
 * Copyright (c) 2019-Present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import 'react-native-gesture-handler';
import React from 'react';
import { SafeAreaView } from 'react-native';
import { isAuthenticated } from '@okta/okta-react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import LoginScreen from './app/LoginScreen.js';
import ProfileScreen from './app/ProfileScreen.js';
import { createConfig } from '@okta/okta-react-native';
import configFile from './samples.config';
import Spinner from 'react-native-loading-spinner-overlay';

const Stack = createStackNavigator();

export default class App extends React.Component {

  constructor() {
    super();

    this.state = {
      progress: true,
      authenticated: false,
    };
    
    this.checkAuthentication = this.checkAuthentication.bind(this);
  }

  async checkAuthentication() {
    const result = await isAuthenticated();

    this.setState({
      authenticated: result.authenticated, 
      progress: false
    });
  }

  async componentDidMount() {
    await createConfig({
      clientId: configFile.oidc.clientId,
      redirectUri: configFile.oidc.redirectUri,
      endSessionRedirectUri: configFile.oidc.endSessionRedirectUri,
      discoveryUri: configFile.oidc.discoveryUri,
      scopes: configFile.oidc.scopes,
      requireHardwareBackedKeyStore:
        configFile.oidc.requireHardwareBackedKeyStore,
    });
    

    await this.checkAuthentication();
  }

  render() {
    if (this.state.progress) {
      return (
        <SafeAreaView>
          <Spinner visible={true} />
        </SafeAreaView>
      );
    }
  
    return (
      <NavigationContainer>
        <Stack.Navigator initialRouteName={this.state.authenticated ? 'Profile' : 'Login'}>
          <Stack.Screen 
            name="Login" 
            component={LoginScreen} 
            options={{ title: 'Login', headerLeft: null }} 
          />
          <Stack.Screen 
            name="Profile" 
            component={ProfileScreen} 
            options={{ title: 'User Profile'}} 
          />
        </Stack.Navigator>
      </NavigationContainer>
    );
  }
}
