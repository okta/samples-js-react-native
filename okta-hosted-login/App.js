import React from 'react';

import { Button, StyleSheet, Text, View } from 'react-native';
import TokenClient from '@okta/okta-react-native';
import config from './.samples.config';

const tokenClient = new TokenClient({
  issuer: config.oidc.issuer,
  client_id: config.oidc.clientId,
  scope: config.oidc.scope,
  redirect_uri: config.oidc.redirectUri
});

export default class App extends React.Component {
  constructor() {
    super()
    this.state = {
      authenticated: false,
      context: null
    }
    this.checkAuthentication = this.checkAuthentication.bind(this);
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.getUser = this.getUser.bind(this);
    this.getMessages = this.getMessages.bind(this);
  }

  async componentDidMount() {
    await this.checkAuthentication();
  }

  async checkAuthentication() {
    const authenticated = await tokenClient.isAuthenticated();
    if (authenticated !== this.state.authenticated) {
      this.setState({ authenticated: authenticated });
    }
  }

  login = async () => {
    await tokenClient.signInWithRedirect();
    this.checkAuthentication();
  }

  logout = async () => {
    await tokenClient.signOut();
    this.checkAuthentication();
    this.setState({context: '' });
  }

  getUser = async () => {
    if (!this.state.authenticated) {
      this.setContext('User has not logged in.');
      return;
    }
    const user = await tokenClient.getUser();
    this.setContext(JSON.stringify(user, null, 4));
  }

  getMessages = async () => {
    fetch(config.resourceServer.messagesUrl, {  
      method: 'GET',
      headers: {
        Authorization: `Bearer ${await tokenClient.getAccessToken()}`,
        'Content-Type': 'application/json',
      }
    })
    .then(response => response.json())
    .then(json => {
      this.setContext(JSON.stringify(json.messages, null, 4));
    })
    .catch(() => {
      const error = `
      Failed to fetch messages. Please verify the following:

      - You've downloaded one of our resource server examples, and it's running on port 8000.

      - Your resource server example is using the same Okta authorization server (issuer) that you have configured this application to use.
      `
      console.warn(error);
      this.setContext('Failed to fetch messages.');
    });
  }

  setContext = (message) => {
    this.setState({
      context: message
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Okta + React Native</Text>
        <View style={styles.button}>
          {this.state.authenticated ?
            <Button
              onPress={ async () => { this.logout() } }
              title="Logout"
            /> :
            <Button
              onPress={ async () => { this.login() } }
              title="Login"
            />
          }
        </View>
        <View style={styles.buttonContainer}>
          <View style={styles.button}>
            <Button
              onPress={ async () => { this.getUser() } }
              title="Profile"
            />
          </View>
          <View style={styles.button}>
            <Button
              onPress={ async () => { this.getMessages() } }
              title="Messages"
            />
          </View>
        </View>
        <Text>{this.state.context}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  button: {
    borderRadius: 15,
    width: '40%',
    height: 40,
    marginTop: 10,
    marginBottom: 10
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#FFFFFF',
    alignItems: 'center',
    paddingTop: 50,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
  },
});
