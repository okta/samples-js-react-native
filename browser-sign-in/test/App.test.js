/*
 * Copyright (c) 2019-Present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { waitForState } from 'enzyme-async-helpers';

import { NativeEventEmitter } from 'react-native';

jest.mock(
  '../node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter',
);

const nativeEmitter = new NativeEventEmitter();

jest
  .mock(
    '../node_modules/react-native/Libraries/Components/StatusBar/StatusBar',
    () => 'StatusBar',
  )
  .mock(
    '../node_modules/react-native/Libraries/Components/ScrollView/ScrollView',
    () => 'ScrollView',
  );

global.fetch = jest
  .fn()
  .mockImplementation(() => {
    const promise = new Promise((resolve, _reject) => {
      resolve({
        json: () => {
          return {
            user: [{ foo: 'foo', bar: 'bar' }],
          };
        },
        ok: true,
      });
    });
    return promise;
  })
  .mockImplementationOnce(() => {
    const promise = new Promise((resolve, _reject) => {
      resolve({
        json: () => {
          return {
            // eslint-disable-next-line camelcase
            userinfo_endpoint: 'dummy_endpoint',
          };
        },
        ok: true,
      });
    });
    return promise;
  });

describe('app setup', () => {
  it('should render without crashing', () => {
    const rendered = renderer.create(<App />).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('should render correctly', () => {
    const rendered = renderer.create(<App />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should initialize with default state', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.state().authenticated).toBe(false);
    expect(wrapper.state().context).toBe(null);
  });

  it('should render login button if not authenticated', () => {
    const wrapper = shallow(<App />);
    const loginButton = wrapper.find('Button').get(0);
    expect(loginButton.props.title).toBe('Login');
  });

  it('should render logout and get user info buttons if authenticated', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({authenticated: true});
    const logoutButton = wrapper.find('Button').get(0);
    const getUserFromIdButton = wrapper.find('Button').get(1);
    const getUserButton = wrapper.find('Button').get(2);
    const getUserFromTokenButton = wrapper.find('Button').get(3);
    const refreshMyTokensButton = wrapper.find('Button').get(4);
    expect(logoutButton.props.title).toBe('Logout');
    expect(getUserFromIdButton.props.title).toBe('Get User From Id Token');
    expect(getUserButton.props.title).toBe('Get User From Request');
    expect(getUserFromTokenButton.props.title).toBe('Get User From Access Token');
    expect(refreshMyTokensButton.props.title).toBe('Refresh Tokens');
  });

  it('should not render login button if authenticated', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({authenticated: true});
    const loginButton = wrapper.find('Button').get(0);
    expect(loginButton.props.title).not.toBe('Login');
  });

  it('should not render logout and get user info buttons if not authenticated', () => {
    const wrapper = shallow(<App />);
    const logoutButton = wrapper.find('Button').get(0);
    const getUserFromIdButton = wrapper.find('Button').get(1);
    const getUserButton = wrapper.find('Button').get(2);
    const getUserFromTokenButton = wrapper.find('Button').get(3);
    const refreshMyTokensButton = wrapper.find('Button').get(4);
    expect(logoutButton.props.title).not.toBe('Logout');
    expect(getUserFromIdButton).toBe(undefined);
    expect(getUserButton).toBe(undefined);
    expect(getUserFromTokenButton).toBe(undefined);
    expect(refreshMyTokensButton).toBe(undefined);
  });
});

describe('authentication flow', () => {
  it('should detect when the user has logged in', async () => {
    const wrapper = shallow(<App />);
    const loginButton = wrapper.find('Button').get(0);
    await loginButton.props.onPress();
    expect(loginButton.props.title).toBe('Login');
    nativeEmitter.emit('signInSuccess');
    expect(wrapper.state().authenticated).toBe(true);
    expect(wrapper.state().context).toBe('Logged in!');
  });

  it('should detect when the user has signed out', async () => {
    const wrapper = shallow(<App />);
    wrapper.setState({authenticated: true});
    const logoutButton = wrapper.find('Button').get(0);
    await logoutButton.props.onPress();
    expect(logoutButton.props.title).toBe('Logout');
    nativeEmitter.emit('signOutSuccess');
    expect(wrapper.state().authenticated).toBe(false);
    expect(wrapper.state().context).toBe('Logged out!');
  });

  it('should return user profile information from id token' , async () => {
    const mockGetIdToken = require('react-native').NativeModules.OktaSdkBridge.getIdToken;
    mockGetIdToken.mockImplementationOnce(() => {
      // id_token returns { a: 'b' }
      return {'id_token': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjoiYiJ9.jiMyrsmD8AoHWeQgmxZ5yq8z0lXS67_QGs52AzC8Ru8'};
    });
    const wrapper = shallow(<App />);
    wrapper.setState({authenticated: true});
    const profileButton = wrapper.find('Button').get(1);
    await profileButton.props.onPress();
    await waitForState(wrapper, state => state.context !== null);
    expect(profileButton.props.title).toBe('Get User From Id Token');
    expect(wrapper.state().context).toContain('User Profile');
  });

  it('should return user profile information from getUser method' , async () => {
    const mockGetUser = require('react-native').NativeModules.OktaSdkBridge.getUser;
    mockGetUser.mockResolvedValue({ 'name': 'fake name' });
    const wrapper = shallow(<App />);
    wrapper.setState({authenticated: true});
    const profileButton = wrapper.find('Button').get(2);
    await profileButton.props.onPress();
    await waitForState(wrapper, state => state.context !== null);
    expect(profileButton.props.title).toBe('Get User From Request');
    expect(wrapper.state().context).toContain('User Profile');
  });

  it('should return user profile information from fetch method' , async () => {
    const mockGetAccessToken = require('react-native').NativeModules.OktaSdkBridge.getAccessToken;
    mockGetAccessToken.mockImplementation(() => {
      return {'access_token': 'dummy_access_token'};
    });
    const wrapper = shallow(<App />);
    wrapper.setState({authenticated: true});
    const profileButton = wrapper.find('Button').get(3);
    await profileButton.props.onPress();
    await waitForState(wrapper, state => state.context !== null);
    expect(profileButton.props.title).toBe('Get User From Access Token');
    expect(wrapper.state().context).toContain('foo');
  });

  it('should return new accessToken, idToken and refreshToken' , async () => {
    const mockRefreshTokens = require('react-native').NativeModules.OktaSdkBridge.refreshTokens;
    mockRefreshTokens.mockImplementation(() => {
      return {'access_token': 'dummy_accessToken',
              'id_token': 'dummy_idToken',
            'refresh_token': 'dummy_refreshToken'};
    });
    const wrapper = shallow(<App />);
    wrapper.setState({authenticated: true});
    const profileButton = wrapper.find('Button').get(4);
    await profileButton.props.onPress();
    await waitForState(wrapper, state => state.context !== null);
    expect(profileButton.props.title).toBe('Refresh Tokens');
    expect(wrapper.state().context).toContain('Successfully refreshed tokens: ');
  });
});
