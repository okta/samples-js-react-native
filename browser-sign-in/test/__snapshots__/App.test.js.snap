// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app setup should render correctly 1`] = `
Array [
  <StatusBar
    barStyle="dark-content"
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0066cc",
          "fontSize": 30,
          "fontWeight": "bold",
          "paddingTop": 40,
          "textAlign": "center",
        }
      }
      testID="titleLabel"
    >
      Okta + React Native
    </Text>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 40,
            "height": 40,
            "marginBottom": 10,
            "marginHorizontal": 10,
            "marginTop": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="loginButton"
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
    <ScrollView
      contentInsetAdjustmentBehavior="automatic"
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <Text
        testID="descriptionBox"
      />
    </ScrollView>
  </RCTSafeAreaView>,
]
`;
